hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -89,12 +91,13 @@ private MemberListSnapshot(int version, LinkedHashMap<UUID, Member> members, UUI
         }
     }
 
-    public ClientClusterServiceImpl(ILogger logger) {
+    public ClientClusterServiceImpl(HazelcastClientInstanceImpl clientInstance, ILogger logger) {
```

Comment:
cyclic dependency? 

To avoid cyclic dependency: Define an interface and pass that to the connection manager and have the cluster service implement that interface.

Reply:
How is this cyclic dependency? we pass the client instance in the tcpClientConnManager as well.

Reply:
HazelcastClientInstanceImpl has a member `ClientClusterServiceImpl clusterService` and now you have `HazelcastClientInstanceImpl ` as the member of the `ClientClusterServiceImpl`, hence is this not cyclic dependency?